website/docs/error-reference.md

```diff
@@ -2746,6 +2746,24 @@ class MyClass < AbstractSerializable
 end
 ```
 
+## 5073
+
+Abstract classes cannot be instantiated by definition. See
+[Abstract Classes and Interfaces](abstract.md) for more information.
+
+```ruby
+class Abstract
+  extend T::Sig
+  extend T::Helpers
+  abstract!
+
+  sig {abstract.void}
+  def foo; end
+end
+
+Abstract.new # error: Attempt to instantiate abstract class `Abstract`
+```
+
```

Comment:
The error reference is used best when it's describing how to unblock the user. (By contrast, the documentation as currently written merely repeats what the error message says.)

Let's add some suggestions:

```suggestion

To fix this error, there are some options:

- If the class which is marked `abstract!` does not actually have any `abstract` methods, simply remove `abstract!` from the class definition to fix the error.
- If the class _does_ have `abstract` methods, find some concrete subclass to call `new` on instead. If the call to `new` is in a test file, you may wish to make a new, test-only subclass of the abstract class. (Depending on the specifics of the test, it may even be possible to simply define all the abstract methods to simply `raise`, so that other aspects of the parent class can be tested.)

```